thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/OnboardResource.java

```diff
@@ -92,9 +94,63 @@ public Long cloneFunctionGetId(@PathParam("id") Long id,
     }
   }
 
+  /**
+   * Copy the Identity of source anomaly function (with srcId) over in total to destination anomaly function (with destId)
+   * Explicit representation: denote source anomaly function with (srcId, srcFunctionName, srcProperties)
+   *                          denote destination anomaly function with (destId, destFunctionName, destProperties)
+   * "copyIdentity" will update destination anomaly function into source anomaly function's identity
+   * After "copyIdentity", the two functions will become:
+   *        (srcId, srcFunctionName, destProperties)
+   *        (destId, destFunctionName, destProperties)
+   * This in fact updates source anomaly function's properties into destination function's properties
+   * @param srcId : the source function with Identity to be copied to
+   * @param destId : the destination function Id that have properties to be copied to source function
+   * @return OK is success
+   */
+  @POST
+  @Path("function/{srcId}/copyIdentity/{destId}")
+  public Response copyIdentity(@PathParam("srcId") @NotNull Long srcId,
+      @PathParam("destId") @NotNull Long destId) {
+    AnomalyFunctionDTO srcAnomalyFunction = anomalyFunctionDAO.findById(srcId);
+    AnomalyFunctionDTO destAnomalyFunction = anomalyFunctionDAO.findById(destId);
+    destAnomalyFunction.setId(srcId);
+    destAnomalyFunction.setFunctionName(srcAnomalyFunction.getFunctionName());
+    anomalyFunctionDAO.update(destAnomalyFunction); // update properties
+    return Response.ok().build();
+  }
 
-  // util functions for clone anomaly functions
+  /**
+   * Clone anomalies from source anomaly function to destination anomaly function in time range
+   * @param srcId : function Id of source anomaly function
+   * @param destId : function Id of destination anomaly function
+   * @param startTimeIso : start time of anomalies to be cloned, time in ISO format ex: 2016-5-23T00:00:00Z
+   * @param endTimeIso : end time of anomalies to be cloned, time in ISO format
+   * @return true if at least one anomaly being cloned
+   * @throws Exception
+   */
+  @POST
+  @Path("function/{srcId}/cloneAnomalies/{destId}")
+  public Response ClonedAnomalies(@PathParam("srcId") @NotNull long srcId,
+      @PathParam("destId") @NotNull long destId,
+      @QueryParam("start") String startTimeIso,
+      @QueryParam("end") String endTimeIso) throws Exception{
+
+    long start = 0;
+    long end = DateTime.now().getMillis();
 
+    if (startTimeIso != null) {
```

Comment:
If the string is invalid, users are not notified. Please send proper error massage to user.